### PR TITLE
Update the README with some basic information

### PR DESCRIPTION
**Pull request by nlarzon**
If this pull request is accepted it will add some basic information in the README on what the provider does and how it works.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,40 @@
...
+The provider is consists of two data sources `flux_install` and `flux_sync` the data sources are corresponding to [fluxv2 manifests](https://pkg.go.dev/github.com/fluxcd/flux2@v0.2.1/pkg/manifestgen)
+
+The data sources are returing `YAML` manifest so a second provider is needed to apply the manifest into the Kubernetes cluster. See example folder.
+
+The `flux_install` generates manifests to install the `fluxv2` components.
+
+```hcl
+# Flux
+data "flux_install" "main" {
+  target_path = "fleet/infra"
```
Comment (stefanprodan) [inline README.md:16]: The `target_path` should match the CLI example = `target_path = "staging-cluster"`

File: README.md [hunk 2]
```diff
@@ -1,2 +1,40 @@
-# terraform-provider-flux
-Terraform provider for Flux v2
+# Terraform provider FluxV2
```
Comment (stefanprodan) [inline README.md:1]: Please use the repo name here `terraform-provider-flux`

File: README.md [hunk 3]
```diff
@@ -1,2 +1,40 @@
...
+  target_path = "fleet/infra"
+  namespace   = "fluxv2"
+}
+```
+
+`flux_sync` the initial source manifest.
+
+```hcl
+data "flux_sync" "main" {
+  target_path = "fleet/infra"
```
Comment (stefanprodan) [inline README.md:25]: Same here `target_path = "staging-cluster"`

**Review by stefanprodan — CHANGES_REQUESTED** [review]
@nlarzon you need to signoff your commit

File: README.md (hunk 1 above)
Comment (nlarzon) [reply README.md]: fixed

File: README.md (hunk 2 above)
Comment (nlarzon) [reply README.md]: done

File: README.md (hunk 3 above)
Comment (nlarzon) [reply README.md]: done

**Comment by stefanprodan** [comment]
@nlarzon please squash your commits into a single one and sign-off the commit, see DCO bot for instruction

File: README.md [hunk 4]
```diff
@@ -1,2 +1,40 @@
...
+```
+
+`flux_sync` the initial source manifest.
+
+```hcl
+data "flux_sync" "main" {
+  target_path = "staging-cluster"
+  url         = "ssh://git@github.com/${var.github_owner}/${var.repository_name}.git"
+}
+```
```
Comment (phillebaba) [inline README.md:28]: I am fine with having an example the README until we get doc generation working with tfplugindocs.

https://github.com/hashicorp/terraform-plugin-docs

File: README.md [hunk 5]
```diff
@@ -1,2 +1,40 @@
...
+
+The data sources are returing `YAML` manifest so a second provider is needed to apply the manifest into the Kubernetes cluster. See example folder.
+
+The `flux_install` generates manifests to install the `fluxv2` components.
+
+```hcl
+# Flux
+data "flux_install" "main" {
+  target_path = "staging-cluster"
+  namespace   = "fluxv2"
```
[the remark this answers is not in the document]
Comment (stefanprodan) [reply README.md:17]: For sure, namespace should be `flux-system` but I don't see why would we put this in the example

File: README.md (hunk 4 above)
Comment (nlarzon) [reply README.md:28]: Are you working on that? You want that into the pipeline?

Comment (phillebaba) [reply README.md:27]: I will get that done tonight, have most of the things in place already.

**Comment by phillebaba** [comment]
@nlarzon could you rebase with the latest changes from main?

**Review by phillebaba — APPROVED** [review]
LGTM :+1: 

We can merge this then continue working on adding to the docs, but at least it is not empty. Or what do you think  @stefanprodan?

**Review by stefanprodan — APPROVED** [review]

**Pull request merged**